fluentnamer/src/main/java/com/azure/autorest/fluent/transformer/SchemaNameNormalization.java

```diff
@@ -371,21 +371,29 @@ private String overrideName(String name) {
             int index = newName.indexOf(entry.getKey());
             if (index >= 0) {
                 int endIndex = index + entry.getKey().length();
-                boolean replace = true;
-                if (index > 0 && isSameCase(newName.charAt(index - 1), newName.charAt(index))) {
-                    replace = false;
-                } else if (endIndex < newName.length() && isSameCase(newName.charAt(endIndex - 1), newName.charAt(endIndex))) {
-                    replace = false;
-                }
-
-                if (replace) {
+                if (!partialMatch(newName, index, endIndex)
+                    // hack, mgmt only allow `preserve-uppercase-max-length` set to 2, so there can't be a partial match
+                    // for all uppercase, e.g. IP. Otherwise, the max uppercase length is 3, which is larger than 2.
+                    // TODO(xiaofei) Remove when we done migrating m4's `preserve-uppercase-max-length`
```

Comment:
I may not get what's here. One example?

Reply:
What I thought was, with `preserve-uppercase-max-length=2`, m4 won't allow `IPRules`(which has uppercase length= 3). We don't have the m4 restriction yet. Thus we can safely assume, when a name has "IP" in it, it's the max uppercase length and the whole word, and we will replace it.

Though through double check, `preserve-uppercase-max-length` seems not working as I assumed.. May need revisit this hack myself..

Reply:
My guess is m4 first breaks `IPRules` to "ip" + "rules", then check max-uppercase. So it be 2, not 3. Not sure.

Reply:
Got it.. Then it seems m4 has different naming override strategy with us..

Reply:
I think so. m4 logic is more complete. My logic here is not intended for uppercase or lowercase and probably not work for short words.

I guess you can ignore ID IP for now.

Reply:
Sure, will revisit it later.